tower-buffer/src/lib.rs

```diff
@@ -42,18 +40,85 @@ pub struct ResponseFuture<T> {
 /// Errors produced by `Buffer`.
 #[derive(Debug)]
 pub enum Error<T> {
+    /// The `Service` call errored.
     Inner(T),
+    /// The underlying `Service` failed.
     Closed,
 }
 
+/// A wrapper that exposes a `Service` (which does not need to be driven) as a `DirectedService` so
+/// that a construct that is *able* to take a `DirectedService` can also take instances of
+/// `Service`.
+pub struct DirectedService<T, Request>
+where
+    T: Service<Request>,
+{
+    inner: T,
+    _marker: PhantomData<Request>,
+}
+
+impl<T, Request> DirectService<Request> for DirectedService<T, Request>
+where
+    T: Service<Request>,
+{
+    type Response = T::Response;
+    type Error = T::Error;
+    type Future = T::Future;
+
+    fn poll_ready(&mut self) -> Result<Async<()>, Self::Error> {
+        self.inner.poll_ready()
+    }
+
+    fn poll_service(&mut self) -> Result<Async<()>, Self::Error> {
+        // TODO: is this the right thing to do?
+        Ok(Async::Ready(()))
+    }
+
+    fn poll_close(&mut self) -> Result<Async<()>, Self::Error> {
+        // TODO: is this the right thing to do?
+        Ok(Async::Ready(()))
+    }
+
+    fn call(&mut self, req: Request) -> Self::Future {
+        self.inner.call(req)
+    }
+}
+
+/// A handle to a `DirectService` that has been spawned elsewhere.
+pub struct HandleTo<T, Request>
```

Comment:
This can be removed by removing `T: Service` fro m`Buffer`. `Buffer` only needs to be generic over the request and the future.

Reply:
It does mean that I end up with a bunch of code that looks like this:
```rust
foo: Buffer<FooRequest, <FooSvc<A> as Service<FooRequest>>::Future>,
```

Reply:
It gets particularly noisy when you have multiple layers of `Service` combinators, especially with #109 because now the `Request` type has to be named multiple type.